### PR TITLE
enh(arch): refactor openid configuration constructor

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to refactor openid configuration constructor

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 1]
```diff
@@ -39,77 +39,103 @@ class OpenIdConfiguration implements ProviderConfigurationInterface
...
-        private ?string $loginClaim,
-        private ?string $clientId,
-        private ?string $clientSecret,
-        private ?string $authenticationType,
-        private bool $verifyPeer,
         private ?ContactTemplate $contactTemplate,
         private bool $isAutoImportEnabled,
         private ?string $emailBindAttribute,
         private ?string $userAliasBindAttribute,
         private ?string $userNameBindAttribute,
```
Comment (callapa) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:137]:
```suggestion
        private bool $isAutoImportEnabled,
        private ?ContactTemplate $contactTemplate = null,
        private ?string $emailBindAttribute = null,
        private ?string $userAliasBindAttribute = null,
        private ?string $userNameBindAttribute = null,
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 2]
```diff
@@ -173,6 +221,33 @@ public function getTrustedClientAddresses(): array
         return $this->trustedClientAddresses;
     }
 
+    /**
+     * @param string[] $trustedClientAddresses
+     * @return self
+     * @throws AssertionException
+     */
+    public function setTrustedClientAddresses(array $trustedClientAddresses): self
+    {
```
Comment (jeremyjaouen) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:230]: empty the property to an empty array before resetting it

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 3]
```diff
@@ -181,6 +256,49 @@ public function getBlacklistClientAddresses(): array
         return $this->blacklistClientAddresses;
     }
 
+    /**
+     * @param string[] $blacklistClientAddresses
+     * @return self
+     */
+    public function setBlacklistClientAddresses(array $blacklistClientAddresses): self
+    {
```
Comment (jeremyjaouen) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:264]: same here

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 4]
```diff
@@ -237,6 +421,30 @@ public function getConnectionScopes(): array
         return $this->connectionScopes;
     }
 
+    /**
+     * @param string[] $connectionScopes
+     * @return self
+     */
+    public function setConnectionScopes(array $connectionScopes): self
+    {
```
Comment (jeremyjaouen) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:429]: same here

**Review by adr-mo — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**